src/ray/core_worker/store_provider/local_plasma_provider.cc

```diff
@@ -76,7 +76,13 @@ Status CoreWorkerLocalPlasmaStoreProvider::Wait(const std::vector<ObjectID> &obj
 
   (*results).resize(object_ids.size());
   for (size_t i = 0; i < object_ids.size(); i++) {
-    (*results)[i] = objects[i]->GetData() != nullptr;
+    if (objects[i] == nullptr) {
+      (*results)[i] = false;
+      continue;
+    }
+
+    (*results)[i] =
+        objects[i]->GetData() != nullptr || objects[i]->GetMetadata() != nullptr;
```

Comment:
I think we can force that either `data` or `metadata` is not empty. And just make `objects[i]` empty instead.

Reply:
sounds good. that should be simpler.